theano/gpuarray/dnn.py

```diff
@@ -1860,6 +1860,8 @@ def local_gpua_pool_dnn_alternative(op, ctx_name, inputs, outputs):
     if not op.ignore_border:
         return
     img, = inputs
+    if len(op.ds) not in (2, 3) or img.ndim != len(op.ds) + 2:
```

Comment:
I think it's needed here because the CPU Pool op supports an arbitrary number of non-pooling dimensions, whereas the GPU ops require exactly two dimensions. Does that make sense or did I misunderstand your comment?
This should also be replaced with a Reshape to two dimensions, or removed altogether if the Pool op only supports 2 non-pooling dimensions.